### ISSUE TITLE
"Error: Input buffer has corrupt header:" for PNG with large XMP metadata

### ISSUE THREAD

**Issue by kevinludwig** [lovell/sharp#1972]
What is the output of running `npx envinfo --binaries --languages --system --utilities`?

npx envinfo --binaries --languages --system --utilities
npx: installed 1 in 2.706s

  System:
    OS: macOS Mojave 10.14.5
    CPU: (8) x64 Intel(R) Core(TM) i7-7920HQ CPU @ 3.10GHz
    Memory: 1.36 GB / 16.00 GB
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 12.11.1 - /usr/local/bin/node
    npm: 6.11.3 - /usr/local/bin/npm
  Utilities:
    Make: 3.81 - /usr/bin/make
    GCC: 10.14. - /usr/bin/gcc
    Git: 2.20.1 - /usr/bin/git
    Clang: 1001.0.46.4 - /usr/bin/clang
    Subversion: 1.10.3 - /usr/bin/svn
  Languages:
    Bash: 3.2.57 - /bin/bash
    Java: 1.8.0_111 - /usr/bin/javac
    Perl: 5.18.4 - /usr/bin/perl
    PHP: 7.1.23 - /usr/bin/php
    Python: 2.7.10 - /usr/bin/python
    Ruby: 2.3.7 - /usr/bin/ruby

What are the steps to reproduce?

Trying to resize a PNG with large XMP metadata, fails with generic "Input buffer has corrupt header". If I use imagemagick `convert -strip` the file it resizes fine. 

What is the expected behaviour?

Resize and ignore/strip the metadata

Are you able to provide a standalone code sample, without other dependencies, that demonstrates this problem?

Are you able to provide a sample image that helps explain the problem?

Not sure if able to provide but the image has ~15MB of XMP metadata and ~4MB of image data. When stripped the 4MB image resizes fine.

**Comment by kevinludwig** [comment]
I did notice just now this closed issue which sounds like the same issue, https://github.com/lovell/sharp/issues/1664, but the resolution seems to imply that this was changed/fixed in sharp 0.23.0, whereas Im running with sharp 0.23.1? Do I mis-understand the resolution on 1664?

**Comment by lovell** [comment]
Please can you share an example image that fails in this manner and the code you're using to process it.

**Comment by kevinludwig** [comment]
Here's the example code

`
const sharp = require('sharp');
const fs = require('fs');

const input = fs.readFileSync('./letsgo.jpg');
sharp(input)
    .resize({width: 400})
    .toFormat('png')
    .then(data => console.log('success, buffer size: ' + data.length))
    .catch(ex => console.log('exception: ' + ex));
`

My production example is a bit more complex because Im streaming from s3 but I did verify that this simplified example with just in memory buffers also fails.

**Comment by kevinludwig** [comment]
```
pngcheck -vt letsgo.png 
File: letsgo.png (12590958 bytes)
  chunk IHDR at offset 0x0000c, length 13
    1400 x 2100 image, 24-bit RGB, non-interlaced
  chunk pHYs at offset 0x00025, length 9: 11811x11811 pixels/meter (300 dpi)
  chunk iCCP at offset 0x0003a, length 2639
    profile name = Photoshop ICC profile, compression method = 0 (deflate)
    compressed profile = 2616 bytes
  chunk iTXt at offset 0x00a95, length 10056557, keyword: XML:com.adobe.xmp
    uncompressed, no language tag
    no translated keyword, 10056536 bytes of UTF-8 text
  chunk cHRM at offset 0x997e0e, length 32
    White x = 0.31269 y = 0.32899,  Red x = 0.63999 y = 0.33001
    Green x = 0.3 y = 0.6,  Blue x = 0.15 y = 0.05999
  chunk IDAT at offset 0x997e3a, length 2531616
    zlib: deflated, 32K window, maximum compression
  chunk IEND at offset 0xc01f66, length 0
No errors detected in letsgo.png (7 chunks, -42.8% compression).
```

**Comment by lovell** [comment]
Ah, you've run into https://github.com/libvips/libvips/pull/1424, which will be fixed in the forthcoming libvips v8.9.0.

**Comment by kevinludwig** [comment]
awesome, thanks for the feedback. Any idea when that may get released?

**Comment by lovell** [comment]
v0.24.0 is now available with a prebuilt libvips v8.9.0. Thanks for reporting this!
